packages/artifact/src/internal-utils.ts

```diff
@@ -97,19 +121,42 @@ export function getArtifactUrl(): string {
  * file systems such as NTFS. To maintain platform-agnostic behavior, all characters that are not supported by an
  * individual filesystem/platform will not be supported on all fileSystems/platforms
  */
-const invalidCharacters = ['\\', '/', '"', ':', '<', '>', '|', '*', '?', ' ']
+const invalidArtifactFileCharacters = ['"', ':', '<', '>', '|', '*', '?', ' ']
+const invalidArtifactNameCharacters = ['\\', '/']
```

Comment:
Is this correct? I would've thought any invalid file character would also be an invalid name character (since the name is the root directory of the artifact)

Reply:
I'll reword some comments and variable names for clarity, the name and path are technically restricted to the set of of characters. The original restrictions comes from `v1` [here]( https://github.com/actions/runner/blob/0cba42590f47d73d811a8bb25823e78a42bd766f/src/Runner.Plugins/Artifact/PublishArtifact.cs#L43)

Previously we would only check the artifact name. The path however that is used for upload was never checked and it could cause problems if illegal characters are included, this would result in the following issue: https://github.com/github/c2c-actions-service/issues/469

Because the upload path can include `/` or `\`, (while the name shouldn't), I split up the character sets